qiskit/compiler/transpile.py

```diff
@@ -201,10 +217,20 @@ def callback_func(**kwargs):
         config = user_config.get_config()
         optimization_level = config.get('transpile_optimization_level', 1)
 
+    if scheduling_method is None:
+        if any_delay_in(circuits):
+            raise TranspilerError("Invalid arguments: When transpiling circuits with delays, "
+                                  "'scheduling_method' is required.")
```

Comment:
what's the rationale for this error? what if I want a delay on one qubit and don't care about the rest of my qubits or how they are scheduled?

Reply:
This error is for notifying users of forgetting to specify `scheduling_method` by mistake. We cannot know `scheduling_method is None` is intended or not. Users often transpile ordinary circuits (intended). If the circuit includes delays, it must not be intended.

Reply:
> If the circuit includes delays, it must not be intended.

I don't think so. A delay is just an instruction. I may want to take a circuit with delays and just optimize it. There should be no error.